pkg/controller/certificates/requestmanager/requestmanager_controller.go

```diff
@@ -395,15 +397,22 @@ func (c *controller) createNewCertificateRequest(ctx context.Context, crt *cmapi
 		},
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(feature.StableCertificateRequestName) {
+		cr.ObjectMeta.GenerateName = ""
+		cr.ObjectMeta.Name = apiutil.DNSSafeShortenTo52Characters(crt.Name) + "-" + fmt.Sprintf("%d", nextRevision)
+	}
+
 	cr, err = c.client.CertmanagerV1().CertificateRequests(cr.Namespace).Create(ctx, cr, metav1.CreateOptions{FieldManager: c.fieldManager})
 	if err != nil {
 		c.recorder.Eventf(crt, corev1.EventTypeWarning, reasonRequestFailed, "Failed to create CertificateRequest: "+err.Error())
 		return err
 	}
 
 	c.recorder.Eventf(crt, corev1.EventTypeNormal, reasonRequested, "Created new CertificateRequest resource %q", cr.Name)
-	if err := c.waitForCertificateRequestToExist(cr.Namespace, cr.Name); err != nil {
-		return fmt.Errorf("failed whilst waiting for CertificateRequest to exist - this may indicate an apiserver running slowly. Request will be retried. %w", err)
+	if !utilfeature.DefaultFeatureGate.Enabled(feature.StableCertificateRequestName) {
```

Comment:
Sorry, just caught this.. perhaps add a quick comment here explaining why we don't call `waitForCertificateRequestToExist` if the feature is enabled 👀

```
// If the StableCertificateRequestName feature gate is enabled, skip waiting for our informer cache/lister to
// observe the creation event and instead rely on an AlreadyExists error being returned if we do attempt a
// CREATE for the same CertificateRequest name again early.
```

Also, could you switch this `if` statement to assert the _positive_, e.g:

```
if utilfeature.DefaultFeatureGate.Enabled(feature.StableCertificateRequestName) {
    return nil
}

// ... existing `waitForCertificateRequestToExist` code here
```

Again, just trying to make it very clear and explicit when anyone looks at this code which behaviour is being gated :)

Reply:
Done!

Reply:
Thanks 🙌 that makes this changeset look very neat/clear that we aren't touching any existing/un-gated code paths 🥳